File: ruby/tirage_terminal.rb
# FR: ouvre le fichier texte_ascii.txt en lecture
# EN: opens the text_ascii.txt file in read mode
fichier_ascii = open("texte_ascii.txt")

# FR: lit le fichier et enregistre le texte
# EN: reads the file and saves the text
texte_ascii = fichier_ascii.read

# FR: affiche le texte ascii pour la présentation
# EN: displays ascii text for presentation
puts texte_ascii

# FR: ouvre et enregistre les utilisateurs dans un tableau
# EN: opens and registers users in a table
tab_utilisateur = open("utilisateur.txt").map(&:chomp)

while 1
    # FR: attend une réponse de l'utilisateur
    # EN: reads the file and saves the users in an array
    puts "- 'GO' pour faire un tirage\n- 'STOP' pour arrêter\n"
    reponse = gets.chomp

    # FR: si 'GO' faire un tirage
    # EN: if 'GO' do a draw
    if reponse == "GO"
        # FR: génère un nombre aléatoire
        # EN: generates a random number
        utilisateur_random = rand(tab_utilisateur.length)

        # FR: affiche l'utiilisateur
        # EN: displays user
        puts tab_utilisateur[utilisateur_random]

    # FR: si 'STOP' sortir de la boucle
    # EN: if 'STOP' exit the loop
    elsif reponse == "STOP"
        break
    end
end*

# FR: la politesse tout de même
# EN: politeness all the same
puts "merci, et à bientôt!"

File: CSharp/terminal/tirage_terminal.cs
// FR: lecture et récupération des textes des fichiers
// FR: reading and retrieving file texts
string texteAscii = File.ReadAllText(@".\texte_ascii.txt");
string[] tab_utilisateur = File.ReadAllLines(@".\utilisateur.txt");

// FR: affiche le texte ascii pour la présentation
// EN: displays ascii text for presentation
Console.WriteLine(texteAscii);

// init Random
Random aleatoire = new Random();

while(true)
{
    // FR: attend une réponse de l'utilisateur
    // EN: reads the file and saves the users in an array
    Console.WriteLine("-'GO' pour faire un tirage\n- 'STOP' pour arrêter\n");
    string reponse = Console.ReadLine();

    // FR: si 'GO' faire un tirage
    // EN: if 'GO' do a draw
    if (reponse == "GO")
    {
        // FR: génère un nombre aléatoire
        // EN: generates a random number
        int entierRandom = aleatoire.Next(tab_utilisateur.Length);

        // FR: affiche l'utiilisateur
        // EN: displays user
        Console.WriteLine("\n"+tab_utilisateur[entierRandom]+"\n");
    }
    // FR: si 'STOP' sortir de la boucle
    // EN: if 'STOP' exit the loop
    else if (reponse == "STOP")
    {
        break;
    }
}
// FR: la politesse tout de même
// EN: politeness all the same
Console.WriteLine("merci, et à bientôt!");